posydon/popsyn/synthetic_population.py

```diff
@@ -230,6 +239,10 @@ def parse(self, S1_state=None, S2_state=None, binary_state=None,
                     df_sel_met = pd.concat([df_sel_met, df_tmp])
                     del df_tmp
 
+            if k != 0:
```

Comment:
Like for PR 104, I'd suggest to use `k>0` instead of `k!=0` when you like have something executed from the second iteration on.